Source/control.cpp

```diff
@@ -2618,6 +2618,7 @@ void DrawGoldSplit(int amount)
 		PrintGameStr(388, 140, tempstr, COL_WHITE);
 	}
 	if (amount > 0) {
+		// BUGFIX: loop condition should be `tempstr[i] != 0`, not `i < tempstr[i]`.

```

Comment:
WTF :joy: 

Should work just fine as long as the string is less then 32 long and only contains one line :man_shrugging:

Reply:
From the land of the goblins and trolls, it becomes very apparent this was a bug.
```c
for (i = 0; i < tempstr[i] != 0; i++)
```